src/dft/backends/mklgpu/commit.cpp

```diff
@@ -53,14 +57,22 @@ class mklgpu_commit final : public dft::detail::commit_impl<prec, dom> {
     // Equivalent MKLGPU precision and domain from OneMKL's precision / domain.
     static constexpr dft::precision mklgpu_prec = to_mklgpu(prec);
     static constexpr dft::domain mklgpu_dom = to_mklgpu(dom);
+
+    // A pair of descriptors are needed because of the [[deprecated]]IN/OUTPUT_STRIDES vs F/BWD_STRIDES API.
+    // Of the pair [0] is fwd DFT, [1] is backward DFT. If possible, the pointers refer to the same desciptor.
+    // Both pointers must be valid.
```

Comment:
I didn't realize using F/BWD_STRIDES would add this much complexity. Would we be able to keep only one descriptor if we required oneAPI 2024.1?
In previous PRs we have discussed that we typically only need to support the latest release: https://github.com/oneapi-src/oneMKL/pull/468#issuecomment-2040116467
I'm concerned of adding complexity for a feature that users may not need.

Reply:
Unfortunately, I couldn't see a nice way to avoid needing two descriptors whilst mixing the INPUT/OUTPUT & FWD/BWD APIs. The only way I could see of doing this is by continuing to use the I/O API and hiding the deprecation warnings somehow, but I don't know if that's possible.

Reply:
Right, I am still wondering if we should completely remove support for `INPUT/OUTPUT STRIDES` and require oneAPI 2024.1 once oneMKL Interface supports `FWD/BWD STRIDES`. I think it would allow to have just one descriptor for MKLGPU and portFFT backends but I don't know the impact on the other backends. I understand this would be a breaking change. I'm curious to know the opinion of @lhuot on the matter.
Note that we could just hide the deprecation warnings with pragma like [here](https://github.com/oneapi-src/oneMKL/blob/1988d4e3bcea2ae8dbba1835dc0adb7d1047b5a6/tests/unit_tests/dft/include/test_common.hpp#L100) if we just want a quick fix. It would be better to find a long term solution.

Reply:
So you are only updating the oneMKL backends to use FWD/BWD_STRIDES, you are not changing the external API of the oneMKL Interfaces library to add FWD/BWD_STRIDES?

Reply:
This PR only changes the MKLGPU backend to use FWD/BWD_STRIDES. We will make the changes to support FWD/BWD_STRIDES in the oneMKL Interface Library interface in another PR.

Reply:
We've agreed this PR is the best solution for now so resolving this comment.